### PR TITLE
CASINT-11183: cleanup duplicate information about ads, categories, and campaigns fields (OpenAPI vs old style RST)

### PR DESCRIPTION
**Pull request by remenska**
* cleaned up old table with explanations per field of an ad, categories, and campaign. Now the `Advertisements`, `Categories` and `Campaigns` pages are very thin. This will prevent double work when updating the API versioning, as the RST and OpenAPI can grow out of sync
* updated where necessary the OpenAPI spec to capture those details we had in the RST table but were missing in OpenAPI (e.g., microTip). Let's keep fields descriptions in one place, and only use freeform documentation when extensive explanation is needed. Take a look at the ad status move as an example. There is no need anymore for a separate page.
* added Oauth2 endpoints (were missing)
* move the Feed Fields page to be under Feeds in TOC rather than top level (was annoying me for too long!)

Screenshot 2024-12-24 at 16 46 17
Screenshot 2024-12-24 at 16 46 39

### CODE REVIEW

File: openapi/categories/responses.yaml [hunk 1]
```diff
@@ -31,7 +31,9 @@ components:
           minimum: 0
           example: 0
         level:
-          description: 'Level at which the category is defined.'
+          description: 'Level at which the category is defined. Although the number of sub levels is not limited, the
+          category tree has up tp three levels of categories (L1-3). The root node is considered special and reside
```
Comment (ChyuWang) [inline openapi/categories/responses.yaml]: up to?

Comment (ChyuWang) [inline openapi/categories/responses.yaml]: L1-L3

File: src/source/ads.rst [hunk 2]
```diff
@@ -1,716 +1,15 @@
...
 .. index:: Ads, Advertisments
 .. _ads:
 
 Advertisements
 ==============
 
 Advertisements or ads are the main resource on the Sellside API. You can
-create, update and delete them, change the status and get a list of your ads.
+create (calling `POST /ad <https://ecg-icas.github.io/icas/openapi/index.html#/Ads/postAd>`_),
+update and delete them (`PUT /ad/{adId} <https://ecg-icas.github.io/icas/openapi/index.html#/Ads/updateAdById>`_
```
Comment (mmmdamin) [inline src/source/ads.rst:9]: i would add `delete` status to openapi doc

File: src/source/ads.rst [hunk 3]
```diff
@@ -1,716 +1,15 @@
...
-ads are allowed to have the same externalId.
-
-.. index:: status
-.. _ad_status:
-
-status
-""""""
-
-Is a valid value from the list of :ref:`ad_status_overview`. The user can set only one
-of the user controlled states *ACTIVE*, *PAUSED* or *DELETED*.
```
Comment (mmmdamin) [inline src/source/ads.rst:149]: i'd add this to openapi

File: src/source/feeds.rst [hunk 4]
```diff
@@ -46,7 +46,6 @@ For instructions on how to create your feed in specific formats, please expand t
 
 
     TSV format follows specification described on `wiki <https://en.wikipedia.org/wiki/Tab-separated_values>`__.
-    Column names used in TSV feed must superset :ref:`column names <feed-details>` specified.
```
Comment (mmmdamin) [inline src/source/feeds.rst:49]: nice spot

**Review by mmmdamin — DISMISSED** [review]

**Pull request merged**